experanto/interpolators.py

```diff
@@ -113,17 +143,85 @@ def interpolate(self, times: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
                 int
             )
             data = self._data[idx]
+        if self.interpolation_mode == "nearest_neighbor":
+            return data, valid
+        elif self.interpolation_mode == "linear":
+            # we are interested to take the data a bit before and after to have better interpolation
+            # if the target time sequence starts / ends with nan
+            if len(idx.shape) == 1:
+                start_idx = int(max(0, idx[0] - self.interp_window))
+                end_idx = int(
+                    min(
+                        idx[-1] + self.interp_window,
+                        np.floor(
+                            (self.valid_interval.end - self.valid_interval.start)
+                            / self.time_delta
+                        ),
+                    )
+                )
+
+                # time is always first dim
+                array = self._data[start_idx:end_idx]
+                orig_times = (
+                    np.arange(start_idx, end_idx) * self.time_delta
+                    + self.valid_interval.start
+                )
+                assert (
+                    array.shape[0] == orig_times.shape[0]
+                ), "times and data should be same length before interpolation"
+                data = linear_interpolate_sequences(
+                    array, orig_times, valid_times, self.keep_nans
+                )
 
-        return data, valid
+            else:
```

Comment:
not sure I get it, what other case than nearest neighbor or linear is there?

Reply:
this `else` is still under `elif self.interpolation_mode == "linear":`
and its
```
if len(idx.shape) == 1:
    ...
else:
```
because the `start_idx` and `end_idx` needs to be taken along dimensions now

Reply:
Oh, I'm stupid. yes makes sense